packages/export/src/nodes/pivot.js

```diff
@@ -0,0 +1,176 @@
+import _ from 'lodash/fp.js'
+import F from 'futil'
+import ContextureClient from 'contexture-client'
+import {
+  getKey,
+  resultsForDrilldown,
+} from 'contexture-client/exampleTypes/pivot.js'
+import { andGroup, runWith, setFilterOnly } from '../utils.js'
+
+// TODO move to shared folder
+let getPageSize = (grouping) => {
+  let pageSizeGetters = {
+    fieldValues: _.getOr(10, 'size'),
+  }
+  return _.getOr(_.noop, grouping.type, pageSizeGetters)(grouping)
+}
+
+export let getGroupingSize = (
+  node,
+  groupingType,
+  cardinalityResult,
+  exportAllPages
+) => {
+  let getNested = _.get(groupingType)
+  let groupingCardinality = 1 // starting with 1 for the total column/row
+
+  F.walk(getNested)((groupResult, index, parents) => {
```

Comment:
This would be cleaner as a reduce. Here's some quick (not tested) ⛳ 

```js
export let getGroupingSize = (node, groupType, countResult, exportAllPages) =>
  F.reduceTree(_.get(groupType))(
    (result, groupResult, index, parents) => {
      let childSize = _.size(_.get(groupType, groupResult))
      let groupSize = _.getOr(childSize, `${groupType}GroupCount`, groupResult)

      if (!exportAllPages) {
        let grouping = _.get([groupType, _.size(parents)], node)
        groupSize = _.min([groupCount, getPageSize(grouping)])
      }

      // TODO else approximate full cardinality by using the average size of the first page of each grouping
      return result + groupSize
    },
    1, // starting with 1 for the total column/row
    countResult,
  )
```

Reply:
It might even be clearer as a `_.sum` on `mapTree`

Reply:
I think you meant `treeToArrayBy` istead of `mapTree`, but it's not really cleaner, so I will go with the reduce
```
export let getGroupingSize = (node, groupType, countResult, exportAllPages) =>
  _.flow(
    F.treeToArrayBy(_.get(groupType))(
      (groupResult, index, parents) => {
        let childSize = _.size(_.getOr(0, groupType, groupResult))
        let groupSize = _.getOr(childSize, `${groupType}GroupCount`, groupResult)

        if (!groupSize) return 0

        if (!exportAllPages) {
          let grouping = _.get([groupType, _.size(parents)], node)
          groupSize = _.min([groupSize, getPageSize(grouping)])
        }

        // TODO else approximate full cardinality by using the average size of the first page of each grouping
        return groupSize
      },
    ),
    _.sum,
    _.add(1) // add 1 for the grand total row/column
  )(countResult)
  ```